CPP_STYLE_GUIDE.md

```diff
@@ -192,40 +192,35 @@ class FancyContainer {
 
 ### Use `nullptr` instead of `NULL` or `0`
 
-What it says in the title.
+Further reading at [ES.47].
 
 ### Ownership and Smart Pointers
 
-"Smart" pointers are classes that act like pointers, e.g.
-by overloading the `*` and `->` operators. Some smart pointer types can be
-used to automate ownership bookkeeping, to ensure these responsibilities are
-met. `std::unique_ptr` is a smart pointer type introduced in C++11, which
-expresses exclusive ownership of a dynamically allocated object; the object
-is deleted when the `std::unique_ptr` goes out of scope. It cannot be
-copied, but can be moved to represent ownership transfer.
-`std::shared_ptr` is a smart pointer type that expresses shared ownership of a
-dynamically allocated object. `std::shared_ptr`s can be copied; ownership
-of the object is shared among all copies, and the object
-is deleted when the last `std::shared_ptr` is destroyed.
-
-Prefer to use `std::unique_ptr` to make ownership
-transfer explicit. For example:
+[R.20]: Use `std::unique_ptr` or `std::shared_ptr` to represent ownership
+[R.21]: Prefer `unique_ptr` over `shared_ptr` unless you need to share ownership
 
+Use `std::unique_ptr` to make ownership transfer explicit. For example:
 ```cpp
 std::unique_ptr<Foo> FooFactory();
 void FooConsumer(std::unique_ptr<Foo> ptr);
 ```
+Since `std::unique_ptr` has only move semantics, passing one by value trasfers
+ownership to the callee and invalidates the instance the caller hold.
 
-Never use `std::auto_ptr`. Instead, use `std::unique_ptr`.
+Avoid `std::auto_ptr`. It was deprecated in C++11 and removed from C++17.
```

Comment:
This seemed sensible in previous form? There’s no reason to use it in our code, and that it’s deprecated/removed is not really of concern if you already know not to use it

Reply:
I did like the phrasing "Never use", also I do believe an explanation why is nice, since it's very short.

Maybe:
```
Don't use `std::auto_ptr`. 
It was deprecated in C++11 if favor of `std::unique_ptr`, and later removed from C++17. `std::unique_ptr` has move semantics instead the of unusual copy semantics of `std::auto_ptr`.
```

As For the ``` Instead, use `std::unique_ptr`.``` part, it's not 100% correct since auto_ptr was more like shared_ptr

Reply:
Actually we should be able to lint for that. I'll see if I can figure out how to.